static/app/components/compactSelect/listBox.tsx

```diff
@@ -256,13 +265,21 @@ function ListBox<Value extends React.Key>({
         });
       }
     },
-  });
+    [
+      focusManager,
+      firstFocusableKey,
+      lastFocusableKey,
+      listState.selectionManager.focusedKey,
+      shouldFocusWrap,
+      listBoxProps,
+    ]
+  );
 
   return (
     <Fragment>
       {filteredItems.length !== 0 && <Separator role="separator" />}
       {filteredItems.length !== 0 && label && <Label {...labelProps}>{label}</Label>}
-      <SelectListBoxWrap {...mergeProps(listBoxProps, keyboardProps)} ref={ref}>
+      <SelectListBoxWrap {...listBoxProps} onKeyDown={onKeyDown} ref={ref}>
```

Comment:
Rather than combine `onKeyDown` callbacks using `mergeProps`, we want to override the one in `listBoxProps` with our own.